### PR TITLE
Update gateway paths to restore spo support

### LINKED ISSUE #4308: SPO loader routes are currently down on Gateway
## Describe the bug
The loader/spo and loader/spo-df paths are currently down as both are trying to generate routerlicious tokens, which should only happen on the loader/fluid path. 

### To Reproduce
1. Go to https://www.r11s-wu2-ppe.prague.office-int.com/loader/spo-df/Testing12345?chaincode=@fluid-example/shared-text@^0.24.0&template=pp.txt and it shows an infinite spinner followed by a timeout
image
2. Locally running the route shows the same result with the following local error as the spo/spo-df tenant ID cannot be found in the tenants supplied by the config, but they are not needed for those routes
image

### Expected behavior
/loader/fluid routes should be using the routerlicious host and access token and loader/spo & loader/spo-df routes should use the SP tokens
Here is the PR that resolves this (https://github.com/microsoft/FluidFramework/pull/4293)

### PR DESCRIPTION
**Pull request by sumedhb1995**
- Adds two separate resolveSpoUrl and resolveR11sUrl functions as Sharepoint uses its own access token whereas Routerlicious uses locally generated one
- getR11sToken is no longer called on the spo/spo-df loader code paths. This was causing those paths to break as the appTenants available from config did not have the spo/spo-df tenant keys available
- Loader routes now pass an isSpoTenantPath boolean to indicate which service factory to use in the controllers
-- loader and loaderFramed routes select on this boolean to pick between the r11s/odsp document service factory in the loader and loaderHost controllers respectively
-- loaderFrs route reuses the loader controller, as before, but now also passes the additional isSpoTenantPath variable to it

### CODE REVIEW

File: server/gateway/src/controllers/loader.ts [hunk 1]
```diff
@@ -56,7 +57,12 @@ export async function initialize(
         async () => Promise.resolve(resolved.tokens.storageToken),
         async () => Promise.resolve(resolved.tokens.socketToken)));
 
-    const tokenProvider = new GatewayTokenProvider(document.location.origin, resolved.url, hostToken, accessToken);
+    const tokenProvider = new GatewayTokenProvider(
+        document.location.origin,
+        resolved.url,
+        hostToken,
+        isSpoTenant(config.tenantId) ? resolved.tokens.storageToken : accessToken,
```
Comment (tanviraumi) [inline server/gateway/src/controllers/loader.ts:64]: Check in routes/loader.ts and send a flag in the controller to indicate whether its a spo tenant.

Comment (tanviraumi) [reply server/gateway/src/controllers/loader.ts:64]: Also you should do a if else check here as well. When its a SPO tenant, all the routerlicious driver code can be skipped

Comment (tanviraumi) [reply server/gateway/src/controllers/loader.ts:64]: So basically do the same thing you do in routes/loader.ts. Use if else to decide which driver you want.

Comment (sumedhb1995) [reply server/gateway/src/controllers/loader.ts]: Loader routes now pass an isSpoTenantPath boolean to indicate which service factory to use in the controllers

**Review by tanviraumi — APPROVED** [review]

**Pull request merged**